arch/sim/src/sim/up_copyfullstate.c

```diff
@@ -41,7 +41,8 @@
  *
  ****************************************************************************/
 
-#if defined(CONFIG_HOST_X86_64) && !defined(CONFIG_SIM_M32)
+#if (defined(CONFIG_HOST_X86_64) && !defined(CONFIG_SIM_M32)) || \
+     defined(CONFIG_HOST_ARM64)
```

Comment:
remove #if/#else/#endif

Reply:
Okay ... hopefully it's not breaking the 32 bit host simulation.

Reply:
we can simply add this:
#ifdef CONFIG_SIM_M32
static_assert(sizeof(unsigned long) == 4));
#endif
If there isn't preprocess error, it should be good.

Reply:
Are there 32bit simulations in the CI, which would detect that?

Reply:
Not working on Linux(sim-02):

    In file included from sim/up_unblocktask.c:35:
    sim/up_unblocktask.c: In function 'up_unblock_task':
    Error: sim/up_unblocktask.c:92:33: error: passing argument 1 of 'up_copyfullstate' from incompatible pointer type [-Werror=incompatible-pointer-types]
       92 |           up_savestate(rtcb->xcp.regs);
          |                        ~~~~~~~~~^~~~~
          |                                 |
          |                                 xcpt_reg_t * {aka unsigned int *}
    sim/up_internal.h:66:45: note: in definition of macro 'up_savestate'
       66 | #define up_savestate(regs) up_copyfullstate(regs, (xcpt_reg_t *)CURRENT_REGS)
          |                                             ^~~~
    sim/up_internal.h:128:38: note: expected 'long unsigned int *' but argument is of type 'xcpt_reg_t *' {aka 'unsigned int *'}
      128 | void up_copyfullstate(unsigned long *dest, unsigned long *src);
          |                       ~~~~~~~~~~~~~~~^~~~

Reply:
ditto